solarforecastarbiter/metrics/calculator.py

```diff
@@ -602,3 +603,123 @@ def _season_from_months(months):
     seasons = np.array(["DJF", "MAM", "JJA", "SON"])
     months = np.asarray(months)
     return seasons[(months // 3) % 4]
+
+
+def _is_deterministic_forecast(proc_fxobs):
+    return (
+        isinstance(proc_fxobs.original.forecast, datamodel.Forecast) and
+        not isinstance(
+            proc_fxobs.original.forecast,
+            (datamodel.ProbabilisticForecast,
+             datamodel.ProbabilisticForecastConstantValue,
+             datamodel.EventForecast)))
```

Comment:
does `type(proc_fxobs.original.forecast) is datamodel.Forecast` give the same result?

```python
In [1]: class A:
   ...:     pass
   ...:

In [2]: class B(A):
   ...:     pass
   ...:

In [3]: o = B()

In [4]: type(o) is B
Out[4]: True

In [5]: type(o) is A
Out[5]: False

In [6]: a = A()

In [7]: type(a) is A
Out[7]: True

In [8]: type(a) is B
Out[8]: False
```

Reply:
yes, although I remember some discussion about using type vs isinstance somewhere else. In this case we probably don't want inheritance to work by default so type is more appropriate